modules/aws/aws.go

```diff
@@ -360,56 +177,47 @@ aws_secret_access_key = %s`,
 		sshKeyText = ""
 	}
 
-	// notify the user
-	r.notify(uid, strings.Join([]string{body, accessKeyText, sshKeyText}, "\n"))
-}
-
-func (r *run) updateUserGroups(uid string) (updated bool) {
-	gresp, err := r.iam.ListGroupsForUser(&iam.ListGroupsForUserInput{
-		UserName: aws.String(uid),
-	})
-	if err != nil || gresp == nil {
-		r.debug("aws %q: groups of user %q not found, needs to be added",
-			r.p.AccountName, uid)
-	}
-	// iterate through the groups and find the missing ones
-	for _, iamgroup := range r.p.IamGroups {
-		found := false
-		for _, group := range gresp.Groups {
-			gname := strings.Replace(awsutil.Prettify(group.GroupName), `"`, ``, -1)
-			if iamgroup == gname {
-				found = true
-			}
-		}
-		if !found {
-			r.addUserToIamGroup(uid, iamgroup)
-			updated = true
+	if awsm.config.NotifyNewUsers {
+		return awsm.Notify(username, strings.Join([]string{body, accessKeyText, sshKeyText}, "\n"), person)
+	} else {
+		fmt.Println("Notify new users disabled, printing output.")
+		fmt.Printf("Created new user: %s\n", username)
+		fmt.Printf("Email body: \n%s\n\n", strings.Join([]string{body, accessKeyText, sshKeyText}, "\n"))
+		eb, err := notifications.EncryptMailBody(awsm.Notifications, []byte(strings.Join([]string{body, accessKeyText, sshKeyText}, "\n")), person)
+		if err != nil {
+			log.Errorf("Error encrypted email body: %s", err)
+			return err
 		}
+		fmt.Printf("Encrypted email body: \n%s", eb)
 	}
-	return
+
+	return nil
 }
 
-func (r *run) addUserToIamGroup(uid, group string) {
-	if !r.Conf.Create {
-		return
-	}
-	if !r.Conf.ApplyChanges {
-		r.debug("aws %q: would have added AWS IAM user %q to group %q",
-			r.p.AccountName, uid, group)
-		return
-	}
-	resp, err := r.iam.AddUserToGroup(&iam.AddUserToGroupInput{
+func (awsm *AWSModule) Reset(username string, person *person_api.Person) error {
+	localUsername := awsm.LDAPUsernameToLocalUsername(username, awsm.config.UsernameMap)
+	return awsm.resetIamUser(localUsername, person)
+}
+
+func (awsm *AWSModule) Delete(username string) error {
+	localUsername := awsm.LDAPUsernameToLocalUsername(username, awsm.config.UsernameMap)
+	return awsm.deleteIamUser(localUsername)
+}
+
+func (awsm *AWSModule) addUserToIamGroup(username, group string) error {
+	resp, err := awsm.iam.AddUserToGroup(&iam.AddUserToGroupInput{
 		GroupName: aws.String(group),
-		UserName:  aws.String(uid),
+		UserName:  aws.String(username),
 	})
 	if err != nil || resp == nil {
-		log.Printf("[error] aws %q: failed to add user %q to group %q: %v",
-			r.p.AccountName, uid, group, err)
+		log.Errorf("aws %q: failed to add user %q to group %q: %v",
+			awsm.config.AccountName, username, group, err)
+		return err
 	}
-	return
+	return nil
 }
 
-func (r *run) removeIamUser(uid string) {
+func (awsm *AWSModule) deleteIamUser(username string) error {
```

Comment:
does `ListAccessKeys` kill their active sessions / deactivate their STS tokens?

Reply:
No, but implementing something to do this in userplex is a really interesting idea. Due to how STS tokens work (they can't be revoked, you have to change the permissions of the role they were created for, etc), this isn't really feasible for this initial rewrite, but I love the idea.